### PR TITLE
40 mean_dice metric

### LINKED ISSUE #40: MVP example for model validation every N training steps
-Depends on task 2
- Minimal evaluation metrics interface (or using ignite.metrics?)
-Event handler for model validation and early stopping

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #40 .

### Description
Add support to compute mean_dice metric during validation or evaluation.

### Status
**Ready**

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### CODE REVIEW

**Comment by Nic-Ma** [comment]
Hi @ericspod , @wyli , @atbenmurray ,

I already added all features of MeanDice metric and unit tests in this PR.
Could you please help review it?
Thanks.

**Comment by Nic-Ma** [comment]
Hi @IsaacYangSLA ,

The CI test failed in multi-gpu test case due to:
_test_multi_gpu (tests.test_parallel_execution.TestParallelExecution) ... Current run is terminating due to exception: CUDA error: out of memory_
Could you please help take a look?
Thanks.

**Comment by wyli** [comment]
@ericspod this PR is ready for review (ci pipeline failed because of a testing server issue, we are working on a fix)

File: monai/utils/to_onehot.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import torch
+import torch.nn.functional as f
+
+
+def to_onehot(data, num_classes: int):
```
Comment (ericspod) [inline monai/utils/to_onehot.py]: Is this not at least partially a duplicate of monai.networks.utils.one_hot? There isn't a need to restrict the first channel to a size of 1, it's conceivable to define a labelling associated with multichannel data with differing labels for each channel.

Comment (wyli) [reply monai/utils/to_onehot.py]: i think in that multichannel label case, each channel might use a different `num_classes`, we'll need an unstack first and then convert each channel individually

**Pull request merged**